lib/msf/core/exploit/remote/rdp.rb

```diff
@@ -1309,18 +1309,18 @@ def cs_core_data(
     client_dig_product_id = Rex::Text.to_unicode(client_dig_product_id[0..32], 'utf-16le')
 
     body = [
-      [version, width, height].pack('<L<S<S'),
+      [version, width, height].pack('L<S<S<'),
       "\x01\xca", # colour depth (8BPP)
       "\x03\xaa", # SASSequence
-      [keyboard, client_build, client_name, keyboard_type].pack('<L<La32*'),
-      [keyboard_type, keyboard_subtype, keyboard_func_key].pack('<L<L<L'),
+      [keyboard, client_build, client_name, keyboard_type].pack('L<L<a32'),
+      [keyboard_type, keyboard_subtype, keyboard_func_key].pack('L<L<L<'),
```

Comment:
Looks like this should be:

```suggestion
      [keyboard, client_build, client_name].pack('L<L<a32'),
      [keyboard_type, keyboard_subtype, keyboard_func_key].pack('L<L<L<'),
```

If you add a reference the method docs as well:

```ruby
# https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/00f1da4a-ee9c-421a-852f-c19f92343d73
  def cs_core_data(
```